src/buffer_manager.ts

```diff
@@ -251,14 +252,20 @@ export class BufferManager implements Disposable {
     private async handleOpenFile(data: EventBusData<"open-file">) {
         const [fileName, close] = data;
         const currEditor = window.activeTextEditor;
-        let doc: TextDocument | undefined;
+        let doc: NotebookDocument | TextDocument | undefined;
         try {
             if (fileName === "__vscode_new__") {
                 doc = await workspace.openTextDocument();
             } else {
                 const normalizedName = fileName.trim();
-                const filePath = this.findPathFromFileName(normalizedName);
-                doc = await workspace.openTextDocument(filePath);
+                let uri = Uri.parse(this.findPathFromFileName(normalizedName));
+                try {
+                    await workspace.fs.stat(uri);
+                } catch {
+                    uri = Uri.from({ scheme: "untitled", path: normalizedName });
+                    doc = await workspace.openNotebookDocument(uri);
```

Comment:
Why notebook?

Reply:
[Limitations](https://github.com/microsoft/vscode/issues/197836) with `TextDocument`, specifically when there is no active workspace. `openNotebookDocument` prompts for a file path when the document is saved and while it returns a `NotebookDocument` it can still be [used](https://github.com/microsoft/vscode/blob/e1a8566a298dcced016d8e16db95c33c270274b4/src/vs/vscode.d.ts#L10764) as a `TextDocument`

Reply:
That's quite a hack. The code should have a brief comment explaning this, with a URL to the upstream issue.